### PR TITLE
feat: add --analyze flag

### LINKED ISSUE #1838: feat: support webpack-bundle-analyzer with CLI
**Is your feature request related to a problem? Please describe.**
We can have an `analyze` flag which invokes webpack-bundle-analyzer as bundle information is crucial limiting size and determining potential bloats

**Describe the solution you'd like**
we can list it in peer dep and use the analyzer which the user has installed instead of shipping out of the box to keep package size low.

**Describe alternatives you've considered**

**Additional context**
/cc @evilebottnawi

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
Feature

**Did you add tests for your changes?**
WIP
**If relevant, did you update the documentation?**
WIP
**Summary**

Invoke `webpack-bundle-analyzer` plugin via `--analyze` flag.

**Does this PR introduce a breaking change?**
No

**Other information**
Fixes #1838

### CODE REVIEW

**Review by evilebottnawi — DISMISSED** [review]
Good work

/cc @webpack/cli-team

**Comment by snitin315** [comment]
Timeout error on windows CI 😞 .

**Comment by evilebottnawi** [comment]
execa sometimes buggy on windows, do you have windows?

**Comment by snitin315** [comment]
Not me,  Anyone else with windows @webpack/cli-team ?

**Review by evenstensberg — APPROVED** [review]

**Review by evilebottnawi — APPROVED** [review]
Good job!

File: packages/webpack-cli/lib/groups/resolveAdvanced.js [hunk 1]
```diff
@@ -28,6 +32,21 @@ const resolveAdvanced = (args) => {
...
+            if (finalOptions.options && finalOptions.options.plugins) {
+                finalOptions.options.plugins.unshift(bundleAnalyzerVal);
+            } else {
+                finalOptions.options.plugins = [bundleAnalyzerVal];
+            }
+        } else {
+            error(`It looks like ${yellow('webpack-bundle-analyzer')} is not installed.`);
+            process.exitCode = 2;
+        }
+    }
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/groups/resolveAdvanced.js:49]: Can we ask to install `webpack-bundle-analyzer` here?

Comment (snitin315) [reply packages/webpack-cli/lib/groups/resolveAdvanced.js:49]: Yes

**Review by evilebottnawi — APPROVED** [review]

**Review by rishabh3112 — APPROVED** [review]

**Pull request merged**